libraries/chain/db_block.cpp

```diff
@@ -746,5 +797,41 @@ bool database::before_last_checkpoint()const
 {
    return (_checkpoints.size() > 0) && (_checkpoints.rbegin()->first >= head_block_num());
 }
+   
+vector< custom_authority_object > database::get_custom_authorities_by_account( account_id_type account ) const
```

Comment:
I think database is ok, that also exposes it to RPC which is very helpful for external scripting

Reply:
Thank you for advice. Will do. 

> Also, returning a vector is a bad idea because it means massive data-copy.

Sure, I can fix it. But I don't think vector is copied in this case because of RVO and rvalue-references features of the C++. But this can be a tricky moment.

Reply:
In the first place, every time when calling this function, when constructing the vector, data gets copied from boost multi-index container.